File: week1/EcomPlatformSearch/Product.java
package week1.EcomPlatformSearch;

public class Product implements Comparable<Product> {
    int productId;
    String productName;
    String category;

    Product(int productId, String productName, String category) {
        this.productId = productId;
        this.productName = productName;
        this.category = category;
    }

    @Override
    public int compareTo(Product other) {
        return this.productName.compareToIgnoreCase(other.productName);
    }

    @Override
    public String toString() {
        return "Product{" + "id=" + productId + ", name='" + productName + "', category='" + category + "'}";
    }
}

File: week1/E-commerce Platform Search Function/Product.cs
// Product.cs
public class Product
{
    public int ProductId { get; set; }
    public string ProductName { get; set; }
    public string Category { get; set; }

    public Product(int productId, string productName, string category)
    {
        ProductId = productId;
        ProductName = productName;
        Category = category;
    }

    public override string ToString()
    {
        return $"Product{{id={ProductId}, name='{ProductName}', category='{Category}'}}";
    }
}
